src/System.Text.Utf8String/System/Text/Utf8Span.CodePointEnumerable.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+namespace System.Text.Utf8
+{
+    partial struct Utf8Span
+    {
+        public ref struct CodePointEnumerable 
+        {
+            private ReadOnlySpan<byte> _buffer;
+
+            public CodePointEnumerable(byte[] bytes, int index, int length)
+            {
+                _buffer = new ReadOnlySpan<byte>(bytes, index, length);
+            }
+
+            public unsafe CodePointEnumerable(ReadOnlySpan<byte> buffer)
+            {
+                _buffer = buffer;
+            }
+
+            public CodePointEnumerator GetEnumerator()
+            {
+                return new CodePointEnumerator(_buffer);
+            }
+
+            public CodePointReverseEnumerator GetReverseEnumerator()
+            {
+                return new CodePointReverseEnumerator(_buffer);
+            }
+
+            public int Count()
```

Comment:
Why do we need it? If we do need it, can it be non-public?

Reply:
This is the implementation of Utf8String's previous Enumerable. I have left it as it was implemented initially. Would it be worth changing it?

Reply:
Enumerating coding units (bytes) does not make much sense. Enumerating code points does make sense, but it's not this enumerator, is it? And so, I would just remove/hide all the enumerators that enumerate bytes (code units)